llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -3092,6 +3092,12 @@ InstructionCost AArch64TTIImpl::getCastInstrCost(unsigned Opcode, Type *Dst,
     return AdjustCost(
         BaseT::getCastInstrCost(Opcode, Dst, Src, CCH, CostKind, I));
 
+  // For the moment we do not have lowering for SVE1-only fptrunc f64->bf16 as
+  // we use fcvtx undef SVE2. Give them invalid costs.
+  if (!ST->hasSVE2() && ISD == ISD::FP_ROUND && SrcTy.isScalableVector() &&
```

Comment:
Might be worth adding `&& !ST->isStreamingSVEAvailable()` because `fcvtx` is also available to streaming functions.